src/components/Api.js

```diff
@@ -0,0 +1,134 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+        this._baseUrl = baseUrl;
+        this._headers = headers;
+    }
+
+    _handleResponse(res) {
+        if (res.ok) {
+          return res.json().catch((err) => {
+            console.error("Error parsing JSON:", err);
+            return Promise.reject("Failed to parse JSON response");
+          });
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      }
+
+
+    getInitialCards() {
+        return fetch(`${this._baseUrl}/cards`, {
+          method: "GET",
+          headers: this._headers,
+        })
+        .then(this._handleResponse)
+        .then((cards) => {
+          console.log("Fetched initial cards:", cards);
+          return cards;
+        })
+        .catch((err) => {
+          console.error("Error fetching initial cards:", err);
+        });
+    }
+
+    getUserInfo() {
+        return fetch(`${this._baseUrl}/users/me`, {
+            method: "GET",
+            headers: this._headers,
+        })
+        .then(this._handleResponse);
```

Comment:
**COULD BE IMPROVED**

You can make a universal method `_request` not to duplicate the `ok` checking, `headers` and `baseUrl`


```js
 _request(endpoint, options = {}) {
    const finalOptions = {
      // add the default headers
      headers: this._headers,
      // spread the other options if provided
      ...options,
    };
      // add the endpoint to the base url
    const url = `${this._baseUrl}${endpoint}`;
    return fetch(url, finalOptions).then(this._handleResponse);
  }

  // usage

  getUserInfo() {
    // pass only the endpoint here
    return this._request("/users/me");
  }

  getInitialCards() {
    // pass only the endpoint here
    return this._request("/cards");
  }

  // and so on



```